### ISSUE TITLE
Rule Proposal: no-v-model-and-value-attrs-allowed

### ISSUE THREAD

**Issue by kb-coder** [vuejs/eslint-plugin-vue#890]
**Please describe what the rule should do:**

This rule should not allow an element to have both a v-model and :value attribute.

**What category should the rule belong to?**

- [ ] Enforces code style
- [X] Warns about a potential error
- [ ] Suggests an alternate way of doing something
- [ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**
```vue
<input type="text"
  v-model="model.first-name"
  :value="model.first-name"
/>

data () {
  return {
    model: {
      first-name: 'Karen'
    }
  }
}

<input type="checkbox"
  v-model="checked"
  :value="isChecked"
/>

data () {
  return {
    checked: true
  }
},
computed: {
  isChecked () {
    return checked
  }
}

```

**Additional context**

We have many new developers using both v-model and value on elements, which seems incorrect. It would be nice to have this rule to check for this.

**Comment by ota-meshi** [comment]
Thank you for this issue.

I like this idea, but I don't think it can be implemented as an ESLint rule because the property bind by v-model are different for each component.
